shared/build.gradle.kts

```diff
@@ -75,6 +75,7 @@ kotlin {
             implementation(libs.io.ktor.client.content.negotiation)
             implementation(libs.io.ktor.ktor.client.serialization)
             implementation(libs.ktor.client.logging)
+            api("dev.gitlive:firebase-auth:1.10.4")
```

Comment:
If we are using this library we don't need my approach. In that case I would move my implementation to a separate auth plugin module . I believe this plugin should do the same to. Lets do that, lets create 2 other modules. We should name them **plugin-auth-gitlive** and **plugin-auth-macao**

Reply:
Listen, This libarary is officially recommended and available for the Kotlin Multiplatform and compose Multiplatform. 
Why we should use the Android Library separably like you are saying? Remove yours Auth Plugin and other stuff, we don't need them. We will use only the Firebase Git Live.

Reply:
Independently of the library being recommended or not. The whole point behind Macao is quite that, creating a plugin ecosystem. Even though if we use this library the most, I want, as an exercise, being able to have 2 different plugin implementations.
These is what the main purpose of Macao is.
I apologize, for timing reasons I haven't been able to layout the project in a way that it looks more obvious. But I will do that tomorrow.
The project will look like this:
The plugin directory inside `shared` module will be moved to its own module named `plugin-api` 
Then `shared` will depend on this module. Then all plugin implementation code currently in shared, will be moved to a separate module.
For example my existing AuthPlugin implementation will be moved to a module named `plugin-auth-macao-firebase` and your implementation using gitlive will go to `plugin-auth-gitlive-firebase`